libraries-3/src/test/java/com/baeldung/jnats/NatsClientLiveTest.java

```diff
@@ -1,115 +1,128 @@
     }
 
-
-    private NatsClient connectClient() {
-        return new NatsClient();
+    @Test
+    public void givenMessageExchangeViaPublishSubscribe_MessagesReceived() throws Exception {
+        try (NatsClient client = connectClient()) {
+            Subscription replySideSubscription = client.subscribeSync("requestSubject");
+            Subscription publishSideSubscription = client.subscribeSync("replyToSubject");
+            client.publishMessageWithReply("requestSubject", "replyToSubject", "hello there");
+
+            Message message = replySideSubscription.nextMessage(TIMEOUT_MILLIS);
+            assertNotNull(message, "No message!");
+            assertEquals("hello there", new String(message.getData()));
+            client.publishMessage(message.getReplyTo(), "hello back");
+
+            message = publishSideSubscription.nextMessage(TIMEOUT_MILLIS);
+            assertNotNull(message, "No message!");
+            assertEquals("hello back", new String(message.getData()));
+        }
     }
 
     @Test
-    public void whenWildCardSubscription_andMatchTopic_MessageReceived() throws Exception {
-
-        NatsClient client = connectClient();
+    public void givenMessageExchangeViaRequestReply_MessagesReceived() throws Exception {
+        try (NatsClient client = connectClient()) {
+            Subscription replySideSubscription = client.subscribeSync("requestSubject");
 
-        SyncSubscription fooSubscription = client.subscribeSync("foo.*");
+            CompletableFuture<Message> future = client.makeRequest("requestSubject", "hello there");
 
-        client.publishMessage("foo.bar", "bar.foo", "hello there");
+            Message message = replySideSubscription.nextMessage(TIMEOUT_MILLIS);
+            assertNotNull(message, "No message!");
+            assertEquals("hello there", new String(message.getData()));
+            client.publishMessage(message.getReplyTo(), "hello back");
 
-        Message message = fooSubscription.nextMessage(200);
-        assertNotNull("No message!", message);
-        assertEquals("hello there", new String(message.getData()));
+            message = future.get(TIMEOUT_MILLIS, TimeUnit.MILLISECONDS);
+            assertNotNull(message, "No message!");
+            assertEquals("hello back", new String(message.getData()));
+        }
     }
 
     @Test
-    public void whenWildCardSubscription_andNotMatchTopic_NoMessageReceived() throws Exception {
-
-        NatsClient client = connectClient();
-
-        SyncSubscription fooSubscription = client.subscribeSync("foo.*");
-
-        client.publishMessage("foo.bar.plop", "bar.foo", "hello there");
-
-        Message message = fooSubscription.nextMessage(200);
-        assertNull("Got message!", message);
-
-
-        SyncSubscription barSubscription = client.subscribeSync("foo.>");
+    public void whenWildCardSubscription_andMatchTopic_MessageReceived() throws Exception {
```

Comment:
Test names still not matching BDD syntax

Reply:
This was an original test/name. I will change.